src/utilities/mpistubs.h

```diff
@@ -55,7 +55,13 @@ extern "C" {
 #define MPI_CHAR            hypre_MPI_CHAR
 #define MPI_LONG            hypre_MPI_LONG
 #define MPI_BYTE            hypre_MPI_BYTE
+#if defined (HYPRE_SINGLE)
```

Comment:
You don't actually need the ifdef stuff here since each of these names is distinct.